### PR TITLE
Freeze rubocop version on 0.50.0

### PR DESCRIPTION
**Pull request by iNecas**
Two reasons:

1. I'm tired of unexpected failres on PRs caused by new comps being
added to Rubocop

2. new rubocop is broken at some checks:

```
An error occurred while Layout/SpaceInsideArrayLiteralBrackets
cop was inspecting
foreman_maintain/test/lib/cli/upgrade_command_test.rb:89:44.
foreman_maintain/test/lib/cli/upgrade_command_test.rb:101:44.
```

### CODE REVIEW

**Comment by swapab** [comment]
:+1:

**Comment by swapab** [comment]
[FYI] Restarted this travis-ci job: https://travis-ci.org/theforeman/foreman_maintain/jobs/350283391

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ if RUBY_VERSION <= '1.8.7'
 end
 
 if RUBY_VERSION >= '2.0'
-  gem 'rubocop'
+  gem 'rubocop', '0.52.1'
```
Comment (swapab) [inline Gemfile]: https://travis-ci.org/theforeman/foreman_maintain/jobs/350283391
I think we can lock it to `<= 0.52.1` ?

Comment (iNecas) [reply Gemfile]: While in general I agree with `<=`, rubocop is fairly specific as once you configure some cop that has not been defined in previous versions, it starts failing with older versions. Therefore I preffer matching against a specific version

**Comment by iNecas** [comment]
I needed to lock the version to 0.50.0, but should be good to go now.

File: .rubocop.yml [hunk 2]
```diff
@@ -74,7 +74,5 @@ Style/FormatStringToken:
 Naming/HeredocDelimiterNaming:
   Enabled: false
 
-# Enable check once issue is resolved:
-# https://github.com/bbatsov/rubocop/issues/4940
-Lint/UnneededRequireStatement:
+Style/RedundantReturn:
```
Comment (swapab) [inline .rubocop.yml]: Can we also mute cop: `FrozenStringLiteralComment`
On my new setup it complains about it in each and every class/module.
```
test/test_helper.rb:1:1: C: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.
```

Comment (iNecas) [reply .rubocop.yml]: I have not seen that myself, but updated.

**Comment by iNecas** [comment]
Addressed all additional comments. Merging to unblock other PRs to get in.

**Pull request merged**